src/main/resources/org/jpeek/metrics/LCOM4.xsl

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+The MIT License (MIT)
+
+Copyright (c) 2017-2018 Yegor Bugayenko
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">
+  <xsl:template match="skeleton">
+    <metric>
+      <xsl:apply-templates select="@*"/>
+      <title>LCOM4</title>
+      <description>
+        <xsl:text>'LCOM4' is a 1995 revision by Martin Hitz and Behzad Montazeri,
+          that further improves upon the revised versions LCOM3 proposed by
+          W. Li and S. Henry in "Maintenance Metrics for the Object Oriented Paradigm".
+          LCOM4 establishes connections between methods not just on shared attributes
+          of the class, but also on whether one method calls the other. That is,
+          methods A and B are said to be related if either both use at least one
+          instance variable in common, OR A calls B, OR B calls A.
+          LCOM4 tweaks the formula for method connections in order to eliminate
+          the distorting effects that 'getters' may have on earlier LCOM variants,
+          and also methods that purely rely on the reuse of other methods without
+          accessing attributes directly (think method overloads where the ones
+          with the least args reuse the ones with the most args).
+        </xsl:text>
+      </description>
+      <xsl:apply-templates select="node()"/>
+    </metric>
+  </xsl:template>
+  <xsl:template match="class">
+    <xsl:variable name="class_fqn" select="replace(string-join(../@id | @id, '.'), '^\.', '')"/>
+    <xsl:variable name="A" select="attributes/attribute[@static = 'false']/text()"/>
+    <xsl:variable name="a" select="count($A)"/>
+    <xsl:variable name="M" select="methods/method"/>
+    <xsl:variable name="m" select="count($M)"/>
+    <!--
+    @todo #8:30min LCOM4: #156 needs to be fixed in order to avoid mistakenly capturing ops
```

Comment:
@llorllale I don't really understand this todo: "to avoid mistakenly capturing ops on fields as if they actually belong to our enclosing class instead of another class" -- which is the "enclosing" class and which is "another" class? Maybe rephrase it?

Reply:
@amihaiemil ok